src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -667,6 +668,40 @@ public override ImmutableArray<string> TupleElementNames
         }
 
         /// <summary>
+        /// Returns the symbols for all the tuple fields (in order and cached).
+        /// </summary>
+        public override ImmutableArray<FieldSymbol> TupleElementFields
+        {
+            get
+            {
+                if (_lazyFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyFields, CreateFields());
+                }
+
+                return _lazyFields;
+            }
+        }
+
+        private ImmutableArray<FieldSymbol> CreateFields()
+        {
+            var builder = ArrayBuilder<FieldSymbol>.GetInstance(_elementTypes.Length);
+            var fields = GetMembers().OfType<FieldSymbol>();
+
+            foreach (var field in fields)
+            {
+                if (field.IsTupleField && field.TupleElementIndex >= 0)
+                {
+                    builder.SetItem(field.TupleElementIndex, field);
```

Comment:
This approach gives no guarantee that we will end up with a builder having the right number of items at the end. In addition, it is hard to assert that we do not have two fields for the same spot. Consider reserving the right number of items at the GetInstance call instead, and using normal indexing. This way we can easily assert that the value is null before we overwrite it. In addition, we will not be able to extend the builder by accident.